packages/ra-core/src/controller/field/useReferenceManyFieldController.ts

```diff
@@ -200,6 +202,73 @@ export const useReferenceManyFieldController = <
         }
     );
 
+    const onSelectAll = useCallback(
```

Comment:
use `useEvent` instead

Reply:
I don't agree, for the same reason that this comment: https://github.com/marmelab/react-admin/pull/10367#discussion_r1880392284